src/eviction.h

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_EVICTION_H
+#define BITCOIN_EVICTION_H
+
+#include <connection_types.h>
+#include <net_permissions.h>
+
+#include <chrono>
+#include <cstdint>
+#include <optional>
+#include <vector>
+
+typedef int64_t NodeId;
+
+struct NodeEvictionCandidate {
+    NodeId id;
+    std::chrono::seconds m_connected;
+    std::chrono::microseconds m_min_ping_time;
+    std::chrono::seconds m_last_block_time;
+    std::chrono::seconds m_last_tx_time;
+    bool fRelevantServices;
+    bool m_relay_txs;
+    bool fBloomFilter;
+    uint64_t nKeyedNetGroup;
+    bool prefer_evict;
+    bool m_is_local;
+    Network m_network;
+    NetPermissionFlags m_perm_flags;
+    ConnectionType m_conn_type;
```

Comment:
Likewise, only one bit of information is actually needed by the eviction logic. Would it be better to replace this with an `m_inbound` boolean?

Perhaps the eviction logic will be changed in future to use more permissions and connection types, but it'll easy enough to change these internal interfaces when that happens.

Reply:
Left this as is because the outbound eviction logic that gets moved to the eviction manager in #25500 does need the connection type.

Of course i could go with your suggestion here and add back the connection type in a later commit for the outbound logic. Let me know if you prefer that.

Reply:
That sounds fine to me.